src/ui/App.tsx

```diff
@@ -0,0 +1,380 @@
+import React, { useState, useEffect } from 'react';
+import { createRoot } from 'react-dom/client';
+import { SemVer } from 'semver';
+import { ErrorMessage } from './ErrorMessage.js';
+import { PackageItem } from './PackageItem.js';
+import { Package, ReleaseType } from './types.js';
+
+type SubmitButtonProps = {
+  isSubmitting: boolean;
+  selections: Record<string, string>;
+  packageDependencyErrors: Record<
+    string,
+    { missingDependentNames: string[]; missingDependencies: string[] }
+  >;
+  onSubmit: () => Promise<void>;
+};
+
+function SubmitButton({
+  isSubmitting,
+  selections,
+  packageDependencyErrors,
+  onSubmit,
+}: SubmitButtonProps) {
+  const isDisabled =
+    isSubmitting ||
+    Object.keys(selections).length === 0 ||
+    Object.keys(packageDependencyErrors).length > 0 ||
+    Object.values(selections).every((value) => value === 'intentionally-skip');
+
+  return (
+    <button
+      onClick={() => void onSubmit()}
+      disabled={isDisabled}
+      className={`mt-6 px-4 py-2 rounded ${
+        isDisabled
+          ? 'bg-blue-300 cursor-not-allowed'
+          : 'bg-blue-500 hover:bg-blue-600'
+      } text-white`}
+    >
+      {isSubmitting
+        ? 'Processing Release (This may take a few minutes)...'
+        : 'Submit Release Selections'}
```

Comment:
Maybe we can be a bit more straightforward about what will happen?

``` suggestion
        ? 'Creating Release Branch (This may take a few minutes)...'
        : 'Create Release Branch'}